cassandra-mesos-executor/src/test/java/io/mesosphere/mesos/frameworks/cassandra/executor/BackupManagerTest.java

```diff
@@ -0,0 +1,191 @@
+package io.mesosphere.mesos.frameworks.cassandra.executor;
+
+import io.mesosphere.mesos.frameworks.cassandra.executor.jmx.JmxConnect;
+import org.apache.cassandra.db.ColumnFamilyStoreMBean;
+import org.apache.cassandra.locator.EndpointSnitchInfoMBean;
+import org.apache.cassandra.service.StorageServiceMBean;
+import org.jetbrains.annotations.NotNull;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.management.RuntimeMXBean;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+
+import static junit.framework.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class BackupManagerTest {
+    public static final String KEYSPACE = "k";
```

Comment:
Modified backup & restore test cases to be more complex. Added test cases for BackupManager internals.